### PR TITLE
#US4R-162: Exposed AFE auto offset removal and us4OEM's HPF configuration in ARRUS.

### PR DESCRIPTION
**Pull request by jrozb91**

### CODE REVIEW

**Comment by pjarosik** [comment]
TBC: Python and MATLAB API

File: arrus/core/api/devices/us4r/Us4R.h [hunk 1]
```diff
@@ -179,6 +179,34 @@ class Us4R : public DeviceWithComponents {
...
+    * See docs of arrus::devices:: for more information.
+    */
+    virtual void disableAfeAutoOffsetRemoval(void) = 0;
+
+    /**
+    * Sets AFE auto offset removal accumulator cycles number.
+    *
+    * See docs of arrus::devices:: for more information.
+    */
+    virtual void setAfeAutoOffsetRemovalCycles(uint16_t cycles) = 0;
```
Comment (PiotrKarwat) [inline arrus/core/api/devices/us4r/Us4R.h:201]: More details, what is the "cycle" "removal delay". What are the units for these values ([s], number of cycles, etc.)?

File: arrus/core/api/devices/us4r/Us4R.h [hunk 2]
```diff
@@ -179,6 +179,34 @@ class Us4R : public DeviceWithComponents {
...
+    * See docs of arrus::devices:: for more information.
+    */
+    virtual void enableAfeAutoOffsetRemoval(void) = 0;
+
+    /**
+    * Disables AFE auto offset removal.
+    *
+    * See docs of arrus::devices:: for more information.
+    */
+    virtual void disableAfeAutoOffsetRemoval(void) = 0;
```
Comment (PiotrKarwat) [inline arrus/core/api/devices/us4r/Us4R.h:194]: What is the default when we dont use the enable/disable AfeAutoOffsetRemoval?

File: arrus/core/api/devices/us4r/Us4R.h (hunk 1 above)
Comment (PiotrKarwat) [reply arrus/core/api/devices/us4r/Us4R.h:201]: What is the default number of cycles (i.e. value that will be used, when this method is not called)?

File: arrus/core/api/devices/us4r/Us4R.h [hunk 3]
```diff
@@ -179,6 +179,34 @@ class Us4R : public DeviceWithComponents {
...
+    * See docs of arrus::devices:: for more information.
+    */
+    virtual void setAfeAutoOffsetRemovalCycles(uint16_t cycles) = 0;
+
+    /**
+    * Sets AFE auto offset removal delay from TX_TRIG.
+    *
+    * See docs of arrus::devices:: for more information.
+    */
+    virtual void setAfeAutoOffsetRemovalDelay(uint16_t delay) = 0;
```
Comment (PiotrKarwat) [inline arrus/core/api/devices/us4r/Us4R.h:208]: What is the default removal delay (i.e. value that will be used, when this method is not called)?

File: arrus/core/api/devices/us4r/Us4R.h (hunk 2 above)
Comment (jrozb91) [reply arrus/core/api/devices/us4r/Us4R.h:192]: Default is disabled, added to comment

File: arrus/core/api/devices/us4r/Us4R.h (hunk 3 above)
Comment (jrozb91) [reply arrus/core/api/devices/us4r/Us4R.h:216]: Default = 0, added to comment

File: arrus/core/api/devices/us4r/Us4R.h (hunk 1 above)
Comment (jrozb91) [reply arrus/core/api/devices/us4r/Us4R.h:210]: Updated comments in Us4R.h. Number of cycles per setting is specified by AFE datasheet and default is 0 = 2047 cycles:
obraz

**Comment by pjarosik** [comment]
@jrozb91 is this PR ready for another review?

**Comment by jrozb91** [comment]
@pjarosik yes

**Pull request closed without merging**